caddyhttp/proxy/reverseproxy.go

```diff
@@ -59,18 +58,20 @@ var (
 )
 
 func createBuffer() interface{} {
-	return make([]byte, 0, 32*1024)
+	b := make([]uint8, 0, 32*1024)
+	return &b
 }
 
 func pooledIoCopy(dst io.Writer, src io.Reader) {
-	buf := bufferPool.Get().([]byte)
+	buf := bufferPool.Get().(*[]uint8)
```

Comment:
uint8? We are copying bytes though. (Yes, they are the same thing roughly, but...)

Reply:
```
caddyhttp/proxy/reverseproxy.go:398:18: argument should be pointer-like to avoid allocations (megacheck)
                        bufferPool.Put(hj.Replay)
                                      ^
```
https://staticcheck.io/docs/checks#SA6002
// If the return value is not a pointer, any conversion from interface{} will 
// involve an allocation. Ensure the return value is a pointer.
New: func() interface{} { b := make([]byte); return &b },
cf:https://go-review.googlesource.com/c/go/+/24371